testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt

```diff
@@ -3,111 +3,133 @@ package com.rakuten.tech.mobile.testapp.ui.settings
+
+    private fun initializeUIComponents() {
+        textInfo = findViewById(R.id.textInfo)
+        edtAppId = findViewById(R.id.editAppId)
+        edtSubscriptionKey = findViewById(R.id.editSubscriptionKey)
+        switchTestMode = findViewById(R.id.switchTestMode)
+        settingsProgressDialog =
+            SettingsProgressDialog(
+                this
+            )
+    }
 
-        edtAppId = settingsDialog.findViewById(R.id.app_id) as EditText
-        edtSubscriptionKey = settingsDialog.findViewById(R.id.subscription_key) as EditText
-        val switchTest = settingsDialog.findViewById<SwitchMaterial>(R.id.switchTest)
+    private fun renderAppSettingsScreen() {
+        textInfo.text = createSettingsInfo()
         edtAppId.setText(settings.appId)
         edtSubscriptionKey.setText(settings.subscriptionKey)
+        switchTestMode.isChecked = settings.isTestMode
+
         edtAppId.addTextChangedListener(settingTextWatcher)
         edtSubscriptionKey.addTextChangedListener(settingTextWatcher)
-        switchTest.isChecked = settings.isTestMode
-        validateSetting()
 
-        renderAppSettingsDialog(settingsDialog, edtAppId, edtSubscriptionKey, switchTest)
-        return true
+        enableSaveView()
     }
 
-    private fun renderAppSettingsDialog(
-        settingsDialog: View,
-        appId: EditText,
-        subscriptionKey: EditText,
-        switchTest: SwitchMaterial
-    ) {
-        val dialog = AlertDialog.Builder(this)
-            .setTitle("${resources.getString(R.string.lb_app_settings)} - Build " +
-                    resources.getString(R.string.miniapp_sdk_version) + " - " +
-                    resources.getString(R.string.build_version))
-            .setView(settingsDialog)
-            .setPositiveButton(R.string.action_save, null)
-            .setNegativeButton(android.R.string.cancel, null)
-            .create()
-
-        dialog.setOnShowListener { _dialog ->
-            btnSave = (_dialog as AlertDialog).getButton(AlertDialog.BUTTON_POSITIVE)
-            btnSave.setOnClickListener {
-                val pb = settingsDialog.findViewById<View>(R.id.pb)
-                pb.visibility = View.VISIBLE
-                updateSettings(appId.text.toString(), subscriptionKey.text.toString(), switchTest.isChecked,
-                    pb, _dialog)
-            }
-
-            _dialog.getButton(AlertDialog.BUTTON_NEGATIVE).setOnClickListener {
-                _dialog.dismiss()
-            }
-        }
-
-        dialog.show()
+    private fun createSettingsInfo(): String {
+        return "Build " + getString(R.string.miniapp_sdk_version) + " - " +
+                getString(R.string.build_version)
     }
 
-    private fun validateSetting() {
-        if (::btnSave.isInitialized && ::edtAppId.isInitialized && ::edtSubscriptionKey.isInitialized) {
-            btnSave.isEnabled =
+    private fun enableSaveView() {
+        if (::edtAppId.isInitialized && ::edtSubscriptionKey.isInitialized) {
```

Comment:
As mentioned above, if you accessed these via `kotlinx.android.synthetic`, then you wouldn't need to check if they're initialized, would you?